### PR TITLE
Canonicalize use_extension label 

### PR DESCRIPTION
**Pull request by fmeum**
Canonicalize the label by adding the current module's repo_name if the
label doesn't specify a repository name. This is necessary as
ModuleExtensionUsages are grouped by the string value of this label, but
later mapped to their Label representation. If multiple strings map to
the same Label, this would result in a crash.

Also enforce that `module()` is called first (if at all).

### CODE REVIEW

**Comment by fmeum** [comment]
Stacked on https://github.com/bazelbuild/bazel/pull/17909

**Comment by fmeum** [comment]
@Wyverald I resolved the conflicts.

File: src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleFileGlobals.java [hunk 1]
```diff
@@ -375,8 +382,25 @@ public void registerToolchains(Sequence<?> toolchainLabels) throws EvalException
...
+    String extensionBzlFile;
+    // This can't change anymore as calling module() after this results in an error.
+    String ownName = module.getRepoName().orElse(module.getName());
+    if (module.getKey().equals(ModuleKey.ROOT) && rawExtensionBzlFile.startsWith("@//")) {
+      extensionBzlFile = "@" + ownName + rawExtensionBzlFile.substring(1);
+    } else if (rawExtensionBzlFile.startsWith("//")) {
+      extensionBzlFile = "@" + ownName + rawExtensionBzlFile;
+    } else {
+      extensionBzlFile = rawExtensionBzlFile;
+    }
```
[the remark this answers is not in the document]
Comment (fmeum) [reply src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleFileGlobals.java]: Done

**Comment by sgowroji** [comment]
Hi @Wyverald, Since I can see that this PR has been approved, please let me know whether I should proceed with importing it.Thanks!

**Review by meteorcloudy — APPROVED** [review]
Thanks!!

**Pull request closed without merging**